src/Streamly/Internal/Data/Stream/Serial.hs

```diff
@@ -8,6 +8,10 @@
 {-# LANGUAGE TypeFamilies              #-}
 {-# LANGUAGE UndecidableInstances      #-} -- XXX
 
+#if MIN_VERSION_base(4,17,0)
```

Comment:
Why is this required?

Reply:
I removed the `CPP` guard but left the `-XTypeOperators` extension.